testsupport/wait/host.go

```diff
@@ -442,7 +459,7 @@ func (a *HostAwaitility) printUserSignupWaitCriterionDiffs(actual *toolchainv1al
 		buf.WriteString("\n----\n")
 		buf.WriteString("diffs:\n")
 		for _, c := range criteria {
-			if !c.Match(actual) {
+			if !c.Match(actual) && c.Diff != nil {
```

Comment:
in which case do we have `c.Diff==nil`?

Reply:
I don't remember now but I only added it because I saw a panic in one of the test runs. I don't think it's always necessary for a criterion to have a Diff so I think it's okay to keep it? WDYT?

Reply:
@xcoulon it looks like this is an example when it panics https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/codeready-toolchain_toolchain-e2e/475/pull-ci-codeready-toolchain-toolchain-e2e-master-e2e/1500820654616743936

Reply:
@rajivnathan and I found the cause of the panic: https://github.com/codeready-toolchain/toolchain-e2e/blob/0c718180c962467d3fc48239d8573dd5344238fb/testsupport/wait/host.go#L419-L430 it doesn't set the Diff param